server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -471,4 +479,41 @@ public boolean doMonitor(ServiceEmitter emitter)
       return true;
     }
   }
+
+  private static class IdentityCheckOverrideSslContextFactory extends SslContextFactory
+  {
+    private final TLSServerConfig tlsServerConfig;
+    private final TLSCertificateChecker certificateChecker;
+
+    public IdentityCheckOverrideSslContextFactory(
+        TLSServerConfig tlsServerConfig,
+        TLSCertificateChecker certificateChecker
+    )
+    {
+      super(false);
+      this.tlsServerConfig = tlsServerConfig;
+      this.certificateChecker = certificateChecker;
+    }
+
+    @Override
+    protected TrustManager[] getTrustManagers(
+        KeyStore trustStore, Collection<? extends CRL> crls
+    ) throws Exception
+    {
+      TrustManager[] trustManagers = super.getTrustManagers(trustStore, crls);
+      TrustManager[] newTrustManagers = new TrustManager[trustManagers.length];
+
+      for (int i = 0; i < trustManagers.length; i++) {
+        if (trustManagers[i] instanceof X509ExtendedTrustManager) {
```

Comment:
Would it be harmful to just leave non `X509ExtendedTrustManager` implementations in this list with an `else` that doesn't wrap them in the `CustomCheckX509TrustManager` and just adds them to `newTrustManagers` as is?